python/tvm/tir/buffer.py

```diff
@@ -176,6 +176,40 @@ def offset_of(self, indices):
         """
         return _ffi_api.BufferOffsetOf(self, indices)  # type: ignore
 
+    def __getitem__(self, indices):
```

Comment:
The logic here is a bit of complicated - we might want to migrate it to c++ instead later